pkg/api/unversioned/group_version.go

```diff
@@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
 
+// GroupVersions can be used to represent a set of desired group versions.
+type GroupVersions []GroupVersion
+
+// VersionForGroupKind identifies the preferred GroupVersion for a GroupKind, or false if no
+// group version is preferred.
+func (gvs GroupVersions) VersionForGroupKind(src GroupKind) (GroupVersion, bool) {
```

Comment:
It's very weird that this takes a group kind but never references the kind. Maybe "PreferredVersionForGroup" or better yet, make PreferredGroup() return a group/version.